fix(isolated-declarations): type of class setter/getter cannot be inferred when the key is a global `Symbol.xxx` expression

Dunqing opened the pull request:
Similar to #11229, using `PropertyKey` to store accessor annotation because `[Symbol.xxx]` is not a static name.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11236?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#11236** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11236?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11236?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#11231** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11231?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#11230** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11230?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#11229** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11229?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11236).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented on crates/oxc_isolated_declarations/src/class.rs:
```diff
@@ -51,12 +48,16 @@ impl<'a> IsolatedDeclarations<'a> {
+    }
+
     pub(crate) fn report_property_key(&self, key: &PropertyKey<'a>) -> bool {
-        if !Self::is_literal_key(key) && !Self::is_global_symbol(key) {
+        if Self::is_valid_property_key(key) {
+            false
+        } else {
```
No difference, and the way is what Rust clippy recommends

Boshen commented:
### Merge activity

* **May 23, 6:49 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(isolated-declarations): type of class setter/getter cannot be inferred when the key is a global `Symbol.xxx` expression (#11236)

Similar to #11229, using `PropertyKey` to store accessor annotation because `[Symbol.xxx]` is not a static name.
